examples/Synthetic Conversation Generation.ipynb

```diff
@@ -0,0 +1,373 @@
+    "            lambda output: output.name == output_name and output.type == \"STRING\",\n",
+    "            result.data.outputs\n",
+    "        )\n",
+    "    ).value\n",
+    "\n",
+    "    return reponse\n",
+    "\n",
+    "\n",
+    "async def get_assistant_response(chat_history: List[types.ChatMessageRequest]) -> str:\n",
+    "    return await invoke_chat_workflow(AI_CHATBOT_WORKFLOW_DEPLOYMENT_NAME, chat_history)\n",
+    "\n",
+    "\n",
+    "async def get_synthetic_user_response(chat_history: List[types.ChatMessageRequest]) -> str:\n",
+    "    return await invoke_chat_workflow(SIMULATED_USER_WORKFLOW_DEPLOYMENT_NAME, chat_history)\n",
+    "\n",
+    "\n",
+    "async def generate_synthetic_conversation(seed_user_message: str, num_conversation_turns: int = DEFAULT_NUM_CONVERSATION_TURNS) -> List[types.ChatMessageRequest]:\n",
+    "    assistant_chat_history: List[types.ChatMessageRequest] = [\n",
+    "        types.ChatMessageRequest(role=\"USER\", text=seed_user_message)\n",
+    "    ]\n",
+    "    synthetic_user_chat_history: List[types.ChatMessageRequest] = [\n",
+    "        types.ChatMessageRequest(role=\"ASSISTANT\", text=seed_user_message)\n",
+    "    ]\n",
+    "    \n",
+    "    for i in range(num_conversation_turns):\n",
+    "        assistant_response = await get_assistant_response(assistant_chat_history)\n",
+    "        \n",
+    "        assistant_chat_history.append(\n",
+    "            types.ChatMessageRequest(role=\"ASSISTANT\", text=assistant_response)\n",
+    "        )\n",
+    "        synthetic_user_chat_history.append(\n",
+    "            types.ChatMessageRequest(role=\"USER\", text=assistant_response)\n",
+    "        )\n",
+    "    \n",
+    "        if i < num_conversation_turns:\n",
+    "            synthetic_user_response = await get_synthetic_user_response(synthetic_user_chat_history)\n",
+    "            \n",
+    "            assistant_chat_history.append(\n",
+    "                types.ChatMessageRequest(role=\"USER\", text=assistant_response)\n",
+    "            )\n",
+    "            synthetic_user_chat_history.append(\n",
+    "                types.ChatMessageRequest(role=\"ASSISTANT\", text=assistant_response)\n",
+    "            )\n",
+    "\n",
+    "    return assistant_chat_history"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0195cce8-a522-41ee-a411-b616b8ef2aed",
+   "metadata": {},
+   "source": [
+    "Here's where we actually invoke our Workflow Deployments. We do so in parallel for faster execution."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "b42176ed-686b-49d5-9267-3d91312fc3ba",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[[ChatMessageRequest(text='What is your refund policy?', role='USER', content=None, source=None),\n",
+       "  ChatMessageRequest(text='What is your refund policy?', role='ASSISTANT', content=None, source=None),\n",
```

Comment:
[q] at the request level long term, do we see ourselves keeping `text` as a proxy to `content={type: STRING, value: str}`?

Reply:
Tbh I haven't thought deeply about it and am open-minded.